src/main/java/sirius/biz/protocol/Protocols.java

```diff
@@ -140,6 +141,43 @@ public void handle(Incident incident) throws Exception {
         }
     }
 
+    private String buildErrorMessages(Throwable throwable) {
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        int numberOfCharactersPerMessage = calcCharactersPerMessage(throwable);
+
+        stringBuilder.append(throwable.getClass().getName()).append(":").append("\n");
+        stringBuilder.append(truncateErrorMessage(throwable.getMessage(), numberOfCharactersPerMessage)).append("\n");
+
+        try {
+            // The first element of the causal chain is always the throwable followed by its cause hierarchy.
+            // Therefore, the first element is skipped.
+            Throwables.getCausalChain(throwable).stream().skip(1).forEach(cause -> {
+                stringBuilder.append("Caused by: ").append(cause.getClass().getName()).append(":").append("\n");
+                stringBuilder.append(truncateErrorMessage(cause.getMessage(), numberOfCharactersPerMessage)).append("\n");
+            });
+        } catch (IllegalArgumentException exception) {
+            // This happens if the causal chain has a circular reference.
+            stringBuilder.append("Warning: Circular reference detected in causal chain. Skipping causes: ")
+                         .append(exception.getMessage());
+        }
+
+        return stringBuilder.toString();
+    }
+
+    private String truncateErrorMessage(String errorMessage, int length) {
+        return Strings.limit(errorMessage, length, true, true, 1000);
+    }
+
+    private int calcCharactersPerMessage(Throwable throwable) {
+        try {
+            return maxMessageLength / Throwables.getCausalChain(throwable).size();
+        } catch (IllegalArgumentException _) {
+            return maxMessageLength;
```

Comment:
Id dont quite understand. 
If this exception should be triggered only the very first exception message will be displayed and thus it can have the whole maxMessageLength

Reply:
Ahhh got it, thanks